src/repository/studyRoomRepository.js

```diff
@@ -0,0 +1,96 @@
+const studyRoomRepository = require('../models/studyRoom.model');
+const { createStudyRoom } = require("../factory/roomFactory");
+const StudyRoom = require('../models/studyRoom.model');
+
+module.exports = class studyRoomRepository {
+    /**
+     * Create a studyRoom.
+     * @param {*} data The body/params of the request. It should contain: owner (email), color, participant, title, description (optional), avatarText (optional).
+     * @returns {StudyRoom} Returns a promise. Resolves with the studyRoom.
+     */
+    static create(data) {
+        return new Promise((resolve, reject) => {
+            const newRoom = createStudyRoom(data)
+            newRoom.save((err, room) => {
+                if (err) { reject(err); }
+                resolve(room);
+            })
+        })
+    }
+
+    /**
+     * Find one event by its studyRoomID.
+     * @param {string} studyRoomID The eventID of the event.
+     * @returns {StudyRoom} Returns a promise. Resolves with a studyRoom.
+     */
+    static findOne(studyRoomID) {
+        return new Promise((resolve, reject) => {
+            StudyRoom.findOne({ studyRoomID: studyRoomID.toString() }).then((room) => {
+                resolve(room);
+            })
+                .catch(err => reject(err))
+        })
+    }
+
+    /**
+     * Delete one studyRoom by its studyRoomID.
+     * @param {string} studyRoomID The studyRoomID of the event.
+     * @returns {number} Returns a promise. Resolves with the number of studyRooms deleted (1 or 0).
+     */
+    static deleteOne(studyRoomID) {
+        return new Promise((resolve, reject) => {
```

Comment:
Some promise returns such as this one seems to be invalid. Please recheck your code to make sure all promises are returned properly. It is also causing a lot of calls (such as those in `room.js`) to not correctly recognize `.then(...)` calls, which leads to backend `console.log(...)` calls to not print anything in the backend console.